### PR TITLE
feat(testing): add Maven coverage support with JaCoCo

### LINKED ISSUE #733: [testing] Add Maven coverage support with JaCoCo
## Problem

The `groovy/getBuildToolInfo` LSP request currently hardcodes coverage support to only return `true` for Gradle projects. Maven projects incorrectly report `supportsCoverage: false` even though Maven supports code coverage via JaCoCo plugin.

## Context

- Identified during implementation of PR #719 (LSP build tool detection)
- The `TestRequestDelegate.kt` explicitly hardcodes: `supportsCoverage = buildTool.name.lowercase() == "gradle"`
- Maven's JaCoCo plugin is the standard for code coverage in Maven projects
- The extension's `CoverageService.ts` currently only handles Gradle JaCoCo output

## Current State

**TestRequestDelegate.kt:**
```kotlin
BuildToolInfo(
    // ...
    // Coverage detection would require additional BuildTool interface changes
    supportsCoverage = buildTool.name.lowercase() == "gradle",
)
```

**index.ts:**
```typescript
// Create coverage service (only works with Gradle for now)
const coverageService = buildTool === 'gradle' ? new CoverageService(logger) : undefined;
```

### Phase 1: LSP Changes

1. Add `getCoverageCommand()` method to `BuildTool` interface
2. Implement in `MavenBuildTool` to generate JaCoCo command
3. Update capability detection to probe for coverage support dynamically

### Phase 2: Extension Changes

1. Update `CoverageService.ts` to handle Maven JaCoCo output
2. Maven JaCoCo output location: `target/site/jacoco/jacoco.xml`
3. Parse the XML format (same as Gradle's JaCoCo)
4. Update `index.ts` to create CoverageService for Maven projects

## Implementation Details

**Maven JaCoCo command:**
```bash
mvn test jacoco:report -Dtest=MyTest
```

**Key files to modify:**
- `groovy-lsp/.../buildtool/BuildTool.kt` - Add getCoverageCommand() interface method
- `groovy-lsp/.../buildtool/maven/MavenBuildTool.kt` - Implement coverage command
- `groovy-lsp/.../testing/TestRequestDelegate.kt` - Dynamic coverage detection
- `editors/code/client/src/features/testing/CoverageService.ts` - Support Maven paths
- `editors/code/client/src/features/testing/index.ts` - Enable for Maven

**JaCoCo output paths:**
- Gradle: `build/reports/jacoco/test/jacocoTestReport.xml`
- Maven: `target/site/jacoco/jacoco.xml`

## References

- JaCoCo Maven Plugin documentation
- PR #719 (LSP build tool detection)
- Current CoverageService implementation

## Acceptance Criteria

- [ ] Maven projects report `supportsCoverage: true` in build tool info
- [ ] Coverage command generation works for Maven test cases
- [ ] CoverageService can parse Maven JaCoCo XML output
- [ ] Coverage decorations appear correctly in editor for Maven projects
- [ ] Integration tests verify Maven coverage workflow end-to-end

### PR DESCRIPTION
**Pull request by albertocavalcante**
Closes #733

---

> [!NOTE]
> Adds unified coverage execution support and improves definition resolution robustness.
> 
> - **Build tools:** Extend `BuildTool` with `getCoverageCommand`; implement for Gradle (`test --tests ... jacocoTestReport`) and Maven (`test -Dtest=... jacoco:report`) with POM JaCoCo checks and warnings
> - **Capabilities:** `TestRequestDelegate` now probes and reports `supportsCoverage` dynamically
> - **Definitions:** Refactor `resolveEffectiveTarget` to accept `uri`/`position` and reject string literal `ConstantExpression`s while still preferring `MethodCallExpression` for method names
> - **Tests:** Add coverage command tests for Gradle/Maven and expand coordinate/position and definition navigation regression tests
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 74175a8123bdd370fe52f7346ef25e2873cbf76f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added coverage command support enabling automatic coverage report generation for test suites in both Gradle and Maven.
  * Implemented dynamic detection of coverage capabilities for build tools.

* **Bug Fixes**
  * String literal targets are no longer treated as navigable in definition resolution, improving accuracy.

* **Tests**
  * Added comprehensive test coverage for coverage command generation and coordinate system boundary conditions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: groovy-build-tool/src/test/kotlin/com/github/albertocavalcante/groovylsp/buildtool/maven/MavenBuildToolTest.kt [hunk 1]
```diff
@@ -20,4 +20,15 @@ class MavenBuildToolTest {
...
+
+    @Test
+    fun `should generate correct coverage command`() {
+        val tool = MavenBuildTool()
+        val project = Paths.get(".")
+
+        val command = tool.getCoverageCommand(project, "com.example.Test", "testMethod")
+
+        assertTrue(command.args.contains("jacoco:report"), "Command should include jacoco:report")
+        assertTrue(command.args.contains("-Dtest=com.example.Test#testMethod"), "Command should target specific test")
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-build-tool/src/test/kotlin/com/github/albertocavalcante/groovylsp/buildtool/maven/MavenBuildToolTest.kt]: Fixed in 00acd978. Added verification for argument order in Maven coverage command.

File: groovy-build-tool/src/test/kotlin/com/github/albertocavalcante/groovylsp/buildtool/gradle/GradleBuildToolTest.kt [hunk 2]
```diff
@@ -74,4 +74,17 @@ class GradleBuildToolTest {
...
+        val tool = GradleBuildTool()
+        val project = Paths.get(".")
+
+        val command = tool.getCoverageCommand(project, "com.example.Test", "testMethod")
+
+        assertTrue(command.args.contains("test"), "Command should include test task")
+        assertTrue(command.args.contains("--tests"), "Command should include --tests flag")
+        assertTrue(command.args.contains("com.example.Test.testMethod"), "Command should target specific test")
+        assertTrue(command.args.contains("jacocoTestReport"), "Command should include jacocoTestReport task")
+    }
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-build-tool/src/test/kotlin/com/github/albertocavalcante/groovylsp/buildtool/gradle/GradleBuildToolTest.kt]: Fixed in 92c413db: Added full suite test case.
